### DISCUSSION TITLE
Saving XYZ of all rendered/ visible points?

### DISCUSSION THREAD

**Discussion by boeboe26** [pyvista/pyvista#3505] [Q&A]
Hi everyone

I know how to extract all depth values that are currently being rendered: https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.get_image_depth.html

But how do I export the xy of these points?

Severin

**Comment by banesullivan** [comment]
You'll need to use hardware selection. We've implemented this in the `enable_cell_picking(through=False)` method on the `Plotter`.

Take a look at https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_cell_picking.html

This method require user interaction but there is  likely a way to automate this such that you could select a camera position and automate the picking. I'll look into automating this and try to report back, but in the meantime, you can manually pick cells in the scene to extract the XYZ coordinates

**Reply by banesullivan** [reply]
Please note that you'll want `through=False` for VISIBLE cell picking. Note that there is a bug in `through=True` as reported in #3317 which I am working to fix

**Comment by boeboe26** [comment]
Thank you for your appreciated answer. 
I also need the correspondence between each pixel in rendered image and the corresponding xyz of the mesh. Maybe it's possible to use ray_trace for that?
